scripts/gen-api-docs.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+APIROOT=$1
+OUTPUTDIR=$2
+PATTERN='^v[0-9]((alpha|beta)[a-z0-9]+)?$'
+
+GOPATH=/Users/$USER/go
+mkdir -p $GOPATH/bin
+GOBIN="$GOPATH/bin"
+
+#Check if 'gen-crd-api-reference-docs' tool is installed, if not, then install. 
+if ! command -v $GOBIN/gen-crd-api-reference-docs  &> /dev/null
+then
+    echo "Installing gen-crd-api-reference-docs in $GOBIN" 
+    #TODO: Replace this with the new release tag.
+    go install github.com/ahmetb/gen-crd-api-reference-docs@v0.3.1-0.20220223025230-af7c5e0048a3
+fi
+
+#Refresh the directory
+rm -rf $OUTPUTDIR
+mkdir $OUTPUTDIR
+
+#Iterate through the directories
+for package in $(find "$APIROOT" -type d); 
+do
+    PACKAGE_VERSION=$(basename $package)
+    GROUPNAME=$(basename $(dirname $package))
+
+    #Filter the main CRD packages
+    if [[ $PACKAGE_VERSION =~ $PATTERN ]] && [[ "$PACKAGE_VERSION" != *"storage" ]] 
```

Comment:
Is this actually needed now given the storage packages also do not have doc.go? (Not entirely sure what doc.go gets the tool so just checking)

Reply:
This checks saves us time in generating docs, if the tool goes through the storage packages then it tries to iterate through all the elements inside the package and then come up with an error `no API packages found in ./api/batch/v1alpha1api20210101storage` which takes a few seconds.

Reply:
Maybe leave a comment explaining that here?